src/server/services/procedures/mortality/mortality.js

```diff
@@ -0,0 +1,228 @@
+        }
+        dict = sub;
+    }
+    return dict;
+}
+
+let CACHED_DATA = undefined;
+async function getData() {
+    if (CACHED_DATA !== undefined) return CACHED_DATA;
+
+    logger.info(`requesting data from ${DATA_SOURCE}`);
+    const resp = await axios({url: DATA_SOURCE, method: 'GET'});
+    if (resp.status !== 200) {
+        logger.error(`download failed with status ${resp.status}`);
+        return {}; // return empty data set on failure
+    }
+    logger.info('download complete - restructuring data');
+
+    const res = {};
+    for (const row of resp.data.split(/\r?\n/)) {
+        const vals = row.split(',');
+        if (vals.length !== 19 || vals[3].trim().length !== 1) continue;
+
+        const country = countryCodeToCountry(vals[0].trim().toUpperCase());
+        const year = parseInt(vals[1]);
+        const week = parseInt(vals[2]);
+        const entry = dictPathOrEmptyInit(res, [country, year, week]);
+
+        const data = {
+            'deaths 0-14': parseFloat(vals[4]),
+            'deaths 15-64': parseFloat(vals[5]),
+            'deaths 65-74': parseFloat(vals[6]),
+            'deaths 75-84': parseFloat(vals[7]),
+            'deaths 85+': parseFloat(vals[8]),
+            'deaths total': parseFloat(vals[9]),
+
+            'rate 0-14': parseFloat(vals[10]),
+            'rate 15-64': parseFloat(vals[11]),
+            'rate 65-74': parseFloat(vals[12]),
+            'rate 75-84': parseFloat(vals[13]),
+            'rate 85+': parseFloat(vals[14]),
+            'rate total': parseFloat(vals[15]),
+        };
+        switch (vals[3].trim().toLowerCase()) {
+        case 'm': entry['male'] = data; break;
+        case 'f': entry['female'] = data; break;
+        case 'b': entry['total'] = data; break;
+        default: logger.warn('unknown gender specifier in raw data source'); break;
+        }
+    }
+
+    logger.info('restructure complete - caching result');
+    CACHED_DATA = res;
+    return res;
+}
+
+const mortality = {};
+
+/**
+ * Get all the mortality data - potentially a lot of data.
+ * Only use this if you truly need access to all data.
+ * This is an object organized by country, then by year, then by week, then broken down by gender.
+ *
+ * @returns {Array}
+ */
+mortality.getEverything = getData;
+
+/**
+ * Gets a list of all the countries represented in the data.
+ * These are not the country names, but a unique identifier for them.
+ *
+ * @returns {Array}
+ */
+mortality.getCountries = async function() {
+    return Object.keys(await getData());
+};
+
+/**
+ * Gets all the data associated with the given country.
+ * This is an object organized by year, then by week, then broken down by gender.
+ * If the specified country is not in the data, returns an empty object (no rows).
+ *
+ * @param {String} country Name of the country to look up
+ * @returns {Array}
+ */
+mortality.getDataForCountry = async function(country) {
+    return (await getData())[country] || {};
```

Comment:
It might be nice to throw an error if the country is not found.

Reply:
Ah, ok. In the new commit I've changed these so that they give back an error message if country does not exist, but still return empty if year and/or week don't exist.